fix: RLS docs, add better explanation of `using` and `with check` expressions (#17225)
supabase/supabase: apps/docs/pages/guides/database/postgres/row-level-security.mdx

for select using ( auth.uid() = user_id );
```

> [comment]: <> (TODO: Improve the RLS docs)
> 

Reviewer: ```suggestion
```

Let's not leave comments in here for readability, if there are outstanding issues to address we can make an issue

create policy "Users can create a profile."
on profiles for insert
to authenticated                    -- the Postgres Role (recommended)
with check ( auth.id() = user_id ); -- the actual Policy
```

### UPDATE Policies

You can specify update policies by combining both the `using` and `with check` expressions.

> The `using` represents the condition that must be true for the update to be allowed, and `with check` ensures that the updates made adhere to the policy constraints.

Reviewer 2: ```suggestion
The `using` clause represents the condition that must be true for the update to be allowed, and `with check` clause ensures that the updates made adhere to the policy constraints.
```

create policy "Users can update their own profile."
on profiles for update
to authenticated                    -- the Postgres Role (recommended)

> using ( auth.id() = user_id )       -- Policy to check the current user's right to update the row.

Reviewer 2: ```suggestion
using ( auth.id() = user_id )       -- checks if the existing row complies with the policy expression
```

> with check ( auth.id() = user_id ); -- Policy to ensure the resultant row still complies with the ownership condition.

Reviewer 2: ```suggestion
with check ( auth.id() = user_id ); -- checks if the new row complies with the policy expression
```
